auth-api/src/auth_api/services/authorization.py

```diff
@@ -109,18 +109,27 @@ def get_user_authorizations_for_entity(business_identifier: str, expanded: bool
                     permissions = PermissionsService.get_permissions_for_membership(auth.status_code, 'SYSTEM')
                     auth_response['roles'] = permissions
         else:
-            keycloak_guid = user_from_context.sub
-            if business_identifier and keycloak_guid:
-                auth = AuthorizationView.find_user_authorization_by_business_number(
-                    business_identifier=business_identifier,
-                    keycloak_guid=keycloak_guid,
-                    org_id=user_from_context.account_id
-                )
+            if business_identifier and (org_id := user_from_context.account_id):
 
-            if auth:
-                permissions = PermissionsService.get_permissions_for_membership(auth.status_code, auth.org_membership)
-                auth_response = Authorization(auth).as_dict(expanded)
-                auth_response['roles'] = permissions
+                # if this is an API GW account, check if the account has access to the resource
+                if user_from_context.login_source == LoginSource.API_GW.value:
+                    auth = AuthorizationView.find_user_authorization_by_business_number(
+                        business_identifier=business_identifier,
+                        org_id=org_id
+                    )
+
+                # Check if the user has access to the resource
+                if keycloak_guid := user_from_context.sub:
```

Comment:
Is it possible to have no sub? I don't think so

Reply:
I agree this should be an elif (api_gw users will have the keycloak_guid and then get nothing). I think the org_id is correct though since user_context is populated from the jwt not the header in the decorator right? I don't see anything in there that tries to get it from the header

Reply:
Might be fine, but still risky.. I'd look at the `account_id_claim` instead of the `account_id`

![image](https://user-images.githubusercontent.com/3484109/197825011-aeaeb3f3-1e60-4410-b1d1-36b4fdb901b5.png)

Reply:
I left the regular user to the account_id, and the gw user to use account_id_claim
I don't think regular users have their account in the JWT as they can change it between calls.